### PR TITLE
LPD-19282 The search only applies to the title of the page

### PR DESCRIPTION
**Pull request by georgel-pop-lr**
Part of story https://liferay.atlassian.net/browse/LPD-4247

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by georgel-pop-lr** [comment]
ci:test:sf

File: modules/apps/layout/layout-impl/src/main/java/com/liferay/layout/internal/search/spi/model/query/contributor/LayoutKeywordQueryContributor.java [hunk 1]
```diff
@@ -34,19 +40,32 @@ public void contribute(
 		SearchContext searchContext =
 			keywordQueryContributorHelper.getSearchContext();
 
-		_queryHelper.addSearchLocalizedTerm(
-			booleanQuery, searchContext, Field.CONTENT, false);
+		List<String> prefixes = new ArrayList<>();
```
Comment (ealonso) [inline modules/apps/layout/layout-impl/src/main/java/com/liferay/layout/internal/search/spi/model/query/contributor/LayoutKeywordQueryContributor.java:43]: @georgel-pop-lr It is better if you do something like:

List<String> prefixes = Arrays.asList(Field.NAME, Field.TITLE);

and then add the;

prefixes.add(Field.CONTENT);

when it is necessary.

File: modules/apps/layout/layout-impl/src/main/java/com/liferay/layout/internal/search/spi/model/query/contributor/LayoutKeywordQueryContributor.java [hunk 2]
```diff
@@ -34,19 +40,32 @@ public void contribute(
...
+
+		Serializable searchOnlyByTitle = searchContext.getAttribute(
+			"searchOnlyByTitle");
+
+		if (Validator.isNotNull(searchOnlyByTitle) &&
+			!searchOnlyByTitle.equals(true)) {
+
+			_queryHelper.addSearchLocalizedTerm(
+				booleanQuery, searchContext, Field.CONTENT, false);
+			prefixes.add(Field.CONTENT);
```
Comment (ealonso) [inline modules/apps/layout/layout-impl/src/main/java/com/liferay/layout/internal/search/spi/model/query/contributor/LayoutKeywordQueryContributor.java:53]: @georgel-pop-lr you should add an empty line here

File: portal-impl/src/com/liferay/portal/service/impl/LayoutLocalServiceImpl.java [hunk 3]
```diff
@@ -2045,12 +2122,22 @@ public int getLayoutsCount(
...
-				groupId, userId, privateLayout, keywords, types, new int[0],
-				QueryUtil.ALL_POS, QueryUtil.ALL_POS, null));
+				groupId, userId, privateLayout, keywords, searchOnlyByTitle,
+				types, new int[0], QueryUtil.ALL_POS, QueryUtil.ALL_POS, null));
 
 		return hits.getLength();
 	}
 
+	@Override
+	public int getLayoutsCount(
```
Comment (ealonso) [inline portal-impl/src/com/liferay/portal/service/impl/LayoutLocalServiceImpl.java:2132]: @georgel-pop-lr I think new methods should be called **search** and **searchCount**, searchOnlyByTitle is a little by weird in a getLayout method

File: modules/apps/layout/layout-impl/src/main/java/com/liferay/layout/internal/search/spi/model/query/contributor/LayoutKeywordQueryContributor.java (hunk 2 above)
Comment (georgel-pop-lr) [reply modules/apps/layout/layout-impl/src/main/java/com/liferay/layout/internal/search/spi/model/query/contributor/LayoutKeywordQueryContributor.java:55]: Do you mean after prefixes.add(Field.CONTENT); line (this empty lines gets removed by the sf) or after }?

Please check how it is now.

File: portal-impl/src/com/liferay/portal/service/impl/LayoutLocalServiceImpl.java (hunk 3 above)
Comment (georgel-pop-lr) [reply portal-impl/src/com/liferay/portal/service/impl/LayoutLocalServiceImpl.java]: Done

File: modules/apps/layout/layout-impl/src/main/java/com/liferay/layout/internal/search/spi/model/query/contributor/LayoutKeywordQueryContributor.java (hunk 1 above)
Comment (georgel-pop-lr) [reply modules/apps/layout/layout-impl/src/main/java/com/liferay/layout/internal/search/spi/model/query/contributor/LayoutKeywordQueryContributor.java]: Done

**Comment by ealonso** [comment]
ci:forward

**Comment by sandrodw3** [comment]
ci:forward

**Comment by georgel-pop-lr** [comment]
ci:report:2548487327

**Comment by georgel-pop-lr** [comment]
ci:test:sf

**Comment by georgel-pop-lr** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:report:2548705466

**Comment by ealonso** [comment]
@georgel-pop-lr there is a unique failure in the report, it is CollectionDisplayWithPagination#NavigateToPagesInCollectionDisplayWithSimplePagination, Could you please verify it?
Thanks!

**Comment by georgel-pop-lr** [comment]
> @georgel-pop-lr there is a unique failure in the report, it is CollectionDisplayWithPagination#NavigateToPagesInCollectionDisplayWithSimplePagination, Could you please verify it? Thanks!

This test is failing on master as well and i don't think it is related to this changes

**Comment by georgel-pop-lr** [comment]
ci:forward:force

**Comment by sandrodw3** [comment]
It fails for me also locally with the same failure, so it's not related to this PR changes

**Pull request closed without merging**